### PR TITLE
feat(eslint-config-fluid): Add plugins / recommended rules for React modules and pre-applies fixes

### PR DESCRIPTION
**Pull request by Josmithr**
BREAKING CHANGE: Note that this is breaking for consumers using React code, as it adds new rules to be enforced.

Plugins being used:
* [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react)
* [eslint-plugin-react-hooks](https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks)

### CODE REVIEW

**Comment by Josmithr** [comment]
These plugins were previously added to a couple of packages individually in:
* #13958
* #13943

File: common/build/eslint-config-fluid/minimal.js [hunk 1]
```diff
@@ -358,6 +356,23 @@ module.exports = {
 				},
 			},
 		},
+		{
+			// Rules only for React files
+			files: ["*.jsx", "*.tsx"],
```
Comment (tylerbutler) [inline common/build/eslint-config-fluid/minimal.js:361]: Is it worth adding a printed config for the settings that apply to react files?

Comment (Josmithr) [reply common/build/eslint-config-fluid/minimal.js:361]: Yeah, that sounds like a good idea.

File: common/build/eslint-config-fluid/package.json [hunk 2]
```diff
@@ -19,9 +19,10 @@
 		"lint:fix": "npm run prettier:fix && npm run eslint:fix",
 		"prettier": "prettier --check .",
 		"prettier:fix": "prettier --write .",
-		"print-config": "concurrently npm:print-config:default npm:print-config:test npm:print-config:minimal npm:print-config:recommended npm:print-config:strict && npm run cleanup-printed-configs",
+		"print-config": "concurrently npm:print-config:* && npm run cleanup-printed-configs",
```
Comment (tylerbutler) [inline common/build/eslint-config-fluid/package.json:22]: Nice, I didn't know this syntax worked! 😍

**Review by tylerbutler — APPROVED** [review]
If you want to release this soon, let me know. I have some other changes I want to coordinate with the next release.

File: common/build/eslint-config-fluid/package.json (hunk 2 above)
Comment (Josmithr) [reply common/build/eslint-config-fluid/package.json:22]: I figured I would give it a shot and was pleasantly surprised 🙂

**Pull request merged**